f-lab-edu/travel-board
# GitHub actions ci

## **[Title]**

- #34 

## **[Context]**

### Github Actions CI workflow 구성을 진행했습니다. 주요 단계는 다음과 같습니다. 
1. build 
   1. setup jdk
   2. setup gradle
   3. cache gradle
   4. test(unit, e2e)
   5. upload-artifact(jar)
2. api-user, api-admin dockerizing and health checking
   1.  download artifact
   2. docker build and run
   3. api health check
   4. docker stop and remove

### 이번 작업을 진행하면서 e2e 테스트 시 testcontainers 오류가 발생했습니다. 
- 상속받은 테스트 클래스 로딩할 때마다 mysql container가 띄워지는 현상 
- 새로운 컨테이너에 새로운 mysql 서버가 띄워지기 때문에 springboot server에서는 connection을 잃게 됩니다.  
- 싱글턴 컨테이너를 사용할 수 있도록 코드 변경했습니다.
- 참고 : [Manual container lifecycle control](https://java.testcontainers.org/test_framework_integration/manual_lifecycle_control/)

## **Checklist**

- [X] base, compare branch가 적절하게 선택되었나요?
- [X] 로컬 환경에서 충분히 테스트 하셨나요?
